src/libponyc/pass/flatten.c

```diff
@@ -40,13 +40,26 @@ static void flatten_isect_element(ast_t* type, ast_t* elem)
   ast_free_unattached(elem);
 }
 
-static ast_result_t flatten_isect(ast_t* ast)
+static ast_result_t flatten_isect(typecheck_t* t, ast_t* ast)
 {
   // Flatten intersections without testing subtyping. This is to preserve any
   // type guarantees that an element in the intersection might make.
```

Comment:
Because:

```ponyc
(Seq[U8] iso & ByteSeq iso)
```

Here, the everything in the ByteSeq union is a subtype of Seq[U8], so Seq[U8] would be removed from the intersection type if we tested subtyping here. But Seq[U8] makes a weaker constructor guarantee (a Seq[U8]) that lets calling `create` on the union type work.